torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -56,6 +57,12 @@ typedef unsigned long long int uint64_t;
 )";
 }
 
+static const std::string& defineComplexTypes() {
+  static std::string result =
+      at::cuda::get_traits_string() + at::cuda::get_complex_body_string();
```

Comment:
This is very helpful and just curious do we have half and bfloat like these too?

Reply:
Yes, they exist in:
https://github.com/csarofeen/pytorch/blob/devel/torch/csrc/jit/codegen/cuda/runtime/fp16_support.cu
https://github.com/csarofeen/pytorch/blob/devel/torch/csrc/jit/codegen/cuda/runtime/bf16_support.cu
IIUC, these files will be converted to strings using the tool in https://github.com/csarofeen/pytorch/blob/devel/torch/csrc/jit/codegen/cuda/tools/stringify_file.py

Reply:
Yes that's the current setup in nvfuser. I was just wondering if we have anything like `at::cuda::get_complex_body_string` for half and bfloat so we could potentially re-use from eager mode. I tried to look a bit but didn't spot any.

Reply:
Ahha, I see what you mean, and we do have something:
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cuda/jit_utils.cpp#L135
I will do some refactor for this part for reuse in later PR.